From b67cf028904e49f4396b0d371da565f80c46d3cd Mon Sep 17 00:00:00 2001
From: Tianhua Sun <tianhua.sun@amlogic.com>
Date: Tue, 21 Apr 2020 18:00:00 +0800
Subject: [PATCH] SurfaceFlinger: set layer blackout for video overlay [1/1]

PD#OTT-9680

Problem:
Video overlay not set layer to black.

Solution:
Set the current layer blackout for video overlay
when draw the layer onto the presentable buffer.

Verify:
Verify on ampere

Change-Id: Ic97e59efaf224d086fef885e6d91c6501b0132e9
Signed-off-by: Tianhua Sun <tianhua.sun@amlogic.com>

diff --git a/services/surfaceflinger/BufferLayer.cpp b/services/surfaceflinger/BufferLayer.cpp
index de62a0dae..698f164a4 100644
--- a/services/surfaceflinger/BufferLayer.cpp
+++ b/services/surfaceflinger/BufferLayer.cpp
@@ -413,7 +413,8 @@ void BufferLayer::onDraw(const RenderArea& renderArea, const Region& clip,
         // is probably going to have something visibly wrong.
     }
 
-    bool blackOutLayer = isProtected() || (isSecure() && !renderArea.isSecure());
+    bool blackOutLayer = isProtected() || (isSecure() && !renderArea.isSecure())
+                         || (isVideoOverlay());
 
     auto& engine(mFlinger->getRenderEngine());
 
-- 
2.26.1

